runtime/src/protocol.rs

```diff
@@ -264,7 +264,14 @@ impl Protocol {
         let mut buffer = vec![0; length];
         reader.read_exact(&mut buffer)?;
 
-        Ok(cbor::from_slice(&buffer)?)
+        let message = cbor::from_slice(&buffer)
+            .map_err(|error| {
+                warn!(self.logger, "Failed to decode message"; "err" => %error);
```

Comment:
Would it be useful to also log the bytes that failed to parse? (maybe in a `debug` level log)

Reply:
Can do that, was not sure. I would definitely not do that in `warn` mode, as this would log up to 16MiB.